apiel/src/parse/mod.rs

```diff
@@ -0,0 +1,55 @@
+pub mod eval;
+pub mod val;
+
+use cfgrammar::Span;
+use lrlex::{lrlex_mod, DefaultLexerTypes};
+use lrpar::{lrpar_mod, Lexeme, Lexer, NonStreamingLexer};
+
+lrlex_mod!("apiel.l");
+lrpar_mod!("apiel.y");
+
+use apiel_y::Expr;
+
+#[cfg(feature = "debug")]
+use lrlex::DefaultLexeme;
+
+pub fn parse_and_evaluate(line: &str) -> Result<Vec<f64>, String> {
+    let lexerdef = apiel_l::lexerdef();
+    let lexer = lexerdef.lexer(line);
+
+    {
+        let mut tokens = String::new();
+        for tok in lexer.iter() {
```

Comment:
You can use name `token` multiple times. If you find it confusing, I'd recommend then using more explicit name like `token_result` or `maybe_token` instead of `tok`

Reply:
I don't mind calling them both `token` :)